rules/rollup_json_test.go

```diff
@@ -0,0 +1,586 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package rules
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/m3db/m3metrics/policy"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRollupTargetView(t *testing.T) {
+	fixture := testRollupTargetJSON("name")
+	expected := RollupTargetView{
+		Name:     "name",
+		Tags:     []string{"tag"},
+		Policies: []policy.Policy{},
+	}
+	require.EqualValues(t, expected, fixture.RollupTargetView())
+}
+
+func TestNewRollupTargetJSON(t *testing.T) {
+	fixture := testRollupTargetView("name")
+	expected := RollupTargetJSON{
+		Name:     "name",
+		Tags:     []string{"tag"},
+		Policies: []policy.Policy{},
+	}
+	require.EqualValues(t, expected, NewRollupTargetJSON(*fixture))
+}
+
+func TestNewRollupRuleJSON(t *testing.T) {
+	targets := []RollupTargetView{
+		*testRollupTargetView("target1"),
+		*testRollupTargetView("target2"),
+	}
+	fixture := testRollupRuleView("rr_id", "rr_name", targets)
+	expected := RollupRuleJSON{
+		ID:     "rr_id",
+		Name:   "rr_name",
+		Filter: "filter",
+		Targets: []RollupTargetJSON{
+			{
+				Name:     "target1",
+				Tags:     []string{"tag"},
+				Policies: []policy.Policy{},
+			},
+			{
+				Name:     "target2",
+				Tags:     []string{"tag"},
+				Policies: []policy.Policy{},
+			},
+		},
+		CutoverMillis:       0,
+		LastUpdatedBy:       "",
+		LastUpdatedAtMillis: 0,
+	}
+	require.EqualValues(t, expected, NewRollupRuleJSON(fixture))
+}
+
+func TestRollupRuleView(t *testing.T) {
+	targets := []RollupTargetJSON{
+		*testRollupTargetJSON("target1"),
+		*testRollupTargetJSON("target2"),
+	}
+	fixture := testRollupRuleJSON("id", "name", targets)
+	expected := &RollupRuleView{
+		ID:     "id",
+		Name:   "name",
+		Filter: "filter",
+		Targets: []RollupTargetView{
+			{
+				Name:     "target1",
+				Tags:     []string{"tag"},
+				Policies: []policy.Policy{},
+			},
+			{
+				Name:     "target2",
+				Tags:     []string{"tag"},
+				Policies: []policy.Policy{},
+			},
+		},
+	}
+	require.EqualValues(t, expected, fixture.RollupRuleView())
+}
+
+func TestRollupRuleSort(t *testing.T) {
```

Comment:
Can you rename this to `TestRollupRuleJSONSort`? Also please do so for the rest of the test functions so the name of the type in the function matches the type you are testing.